### DISCUSSION TITLE
Tab Stacks

### DISCUSSION THREAD

**Discussion by Lu9-ST** [Floorp-Projects/Floorp#1120] [Feature Request]
The feature first introduced by Vivaldi Browser. Group many tabs together into one tab, it can be displayed as divided spaces on top of the tab, as a folded tab that you can unfold, or as a second line of tabs under the main tab bar.

 I always thought of Floorp as a potential open-source and Firefox-based successor to Vivaldi as it already has almost everything from it. Would that be a possibility?

Edit: I realized this post went into _Discussions_ instead of _Issues_, and there are similar posts about this feature, sorry.

**Comment by Ryosuke-Asano** [comment]
Currently, Floorp has Workspaces only due to compatibility with extensions. In the past, I tried to implement tab stack instead of workspaces, but that was failed because of that reason.

**Comment by ttaylor-ion** [comment]
This feature has been added to Floorp as of v12.17.0!
Super excited and finally dropping Vivaldi for good!

But that said... I can see some enhancements made to this incredible feature:
- Add option to "Open links that request a new window" for tab stacks
    - This option would convert the current tab into a stack and add the new tab there, instead of opening links in new tabs
- Unnamed/auto-named tab stacks
    - This option would use the name of the stack's last/current active tab
- Improve readability of tab titles in a tab stack
    - This seems to be related to dynamic background colors based on the active tab.
    - This can be reproduced by creating a tab stack with the active tab being the Floorp website's home page. (Not sure if this is settings dependent.)
